apps/src/music/views/SoundsPanel.tsx

```diff
@@ -158,50 +195,135 @@ const SoundsPanel: React.FunctionComponent<SoundsPanelProps> = ({
   onPreview,
 }) => {
   const folders = library.getAllowedSounds(undefined);
-  const libraryGroupPath = library.groups[0].path;
-
-  // Generate a flat list of entries to render.  We need a flat list because
-  // we will make the headers sticky.
-  const entries: {
-    type: 'folder' | 'sound';
-    folder: SoundFolder;
-    sound?: SoundData;
-  }[] = [];
-  folders.forEach(folder => {
-    entries.push({type: 'folder', folder: folder});
-    folder.sounds.forEach(sound => {
-      entries.push({type: 'sound', folder: folder, sound: sound});
+  const libraryGroupPath = library.libraryJson.path;
+
+  const [selectedFolder, setSelectedFolder] = useState<SoundFolder>(
+    library.getFolderForSoundId(currentValue) || folders[0]
+  );
+  const [mode, setMode] = useState<string>('packs');
+  const [filter, setFilter] = useState<string>('all');
+
+  const currentFolderRef: React.MutableRefObject<HTMLDivElement | null> =
+    useRef(null);
+  const currentSoundRef: React.MutableRefObject<HTMLDivElement | null> =
```

Comment:
minor: might be cleaner to parameterize the `useRef` function call instead of typing the variable, since the type can be inferred that way, i.e. - `const currentFolderRef = useRef<HTMLDivElement>(null)`

Reply:
There were some issues with making that work, and I didn't find a really crisp solution online.  I see we use this technique [here](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/componentLibrary/checkboxDropdown/CheckboxDropdown.tsx#L78-L79).  Maybe @levadadenys has thoughts?

Reply:
Oh interesting - we do something similar in Timeline.tsx, [here](https://github.com/code-dot-org/code-dot-org/blob/07d70ff3ecca702fc76b587bf447bd8ffffb8dc1/apps/src/music/views/Timeline.tsx#L59C23-L59C29) but I could be missing something. Not a huge deal either way though!

Reply:
@breville code you've mentioned is now stored [here](https://github.com/code-dot-org/code-dot-org/blob/1f59a11a940e1cf79d8373a934de9db74b91651e/apps/src/componentLibrary/dropdown/_CustomDropdown.tsx#L59).
First I used your approach, then tried what @sanchitmalhotra126 suggested and everything worked fine out of the box. Might depend on specific use case scenarios though